packages/backend-app-api/src/services/implementations/auth/plugin/keys/StaticConfigPluginKeySource.ts

```diff
@@ -0,0 +1,177 @@
+ *
+ * @remarks
+ *
+ * You can generate a public and private key pair, using
+ * openssl:
+ *
+ * Generate a private key using the ES256 algorithm
+ * ```sh
+ * openssl ecparam -name prime256v1 -genkey -out private.ec.key
+ * ```
+ * Convert it to PKCS#8 format
+ * ```sh
+ * openssl pkcs8 -topk8 -inform PEM -outform PEM -nocrypt -in private.ec.key -out private.key
+ * ```
+ * Extract the public key
+ * ```sh
+ * openssl ec -inform PEM -outform PEM -pubout -in private.key -out public.key
+ * ```
+ *
+ * Provide the paths to private.key and public.key as the respective
+ * private and public key paths in the `create` method.
+ */
+export class StaticConfigPluginKeySource implements PluginKeySource {
+  private constructor(
+    private readonly keyPairs: KeyPair[],
+    private readonly keyDurationSeconds: number,
+  ) {}
+
+  public static async create(options: {
+    sourceConfig: Config;
+    keyDuration: HumanDuration;
+  }): Promise<PluginKeySource> {
+    const keyConfigs = options.sourceConfig
+      .getConfigArray('static.keys')
+      .map(c => {
+        const staticKeyConfig: StaticKeyConfig = {
+          publicKeyFile: c.getString('publicKeyFile'),
+          privateKeyFile: c.getString('privateKeyFile'),
+          keyId: c.getString('keyId'),
+          algorithm: c.getOptionalString('algorithm') ?? DEFAULT_ALGORITHM,
+        };
+
+        return staticKeyConfig;
+      });
+
+    const keyPairs = await Promise.all(
+      keyConfigs.map(async k => await this.loadKeyPair(k)),
+    );
+
+    if (keyPairs.length < 1) {
+      throw new Error(
+        'At least one key pair must be provided in static.keys, when the static key store type is used',
+      );
+    }
+
+    return new StaticConfigPluginKeySource(
+      keyPairs,
+      durationToMilliseconds(options.keyDuration) / SECONDS_IN_MS,
+    );
+  }
+
+  async getPrivateSigningKey(): Promise<JWK> {
+    return this.keyPairs[0].privateKey;
+  }
+
+  async listKeys(): Promise<{ keys: KeyPayload[] }> {
+    const keys = this.keyPairs.map(k => this.keyPairToStoredKey(k));
+    return { keys };
+  }
+
+  private static async loadKeyPair(options: StaticKeyConfig): Promise<KeyPair> {
+    const algorithm = options.algorithm;
+    const keyId = options.keyId;
+    const publicKey = await this.loadPublicKeyFromFile(
+      options.publicKeyFile,
+      keyId,
+      algorithm,
+    );
+    const privateKey = await this.loadPrivateKeyFromFile(
+      options.privateKeyFile,
+      keyId,
+      algorithm,
+    );
+
+    return { publicKey, privateKey, keyId };
+  }
+
+  private static async loadPublicKeyFromFile(
+    path: string,
+    keyId: string,
+    algorithm: string,
+  ): Promise<JWK> {
+    return this.loadKeyFromFile(path, keyId, algorithm, importSPKI);
+  }
+
+  private static async loadPrivateKeyFromFile(
+    path: string,
+    keyId: string,
+    algorithm: string,
+  ): Promise<JWK> {
+    return this.loadKeyFromFile(path, keyId, algorithm, importPKCS8);
+  }
```

Comment:
These seem a bit redundant tbh

Reply:
It's all just copypasta from https://github.com/backstage/backstage/blob/aeb5a2bf68240188b7fd463fa45c35539a5bb7b2/plugins/auth-backend/src/identity/StaticKeyStore.ts and we didn't want to tear it apart and refactor at this point 😅